### PR TITLE
ghc, haskell-infra: #40929 take 2

### PR DESCRIPTION
**Pull request by Ericson2314**
###### Motivation for this change

@peti Pretty sure I fixed the paxmark issues, but there could be other issues.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/compilers/ghc/8.0.2.nix [hunk 1]
```diff
@@ -151,10 +164,11 @@ stdenv.mkDerivation rec {
   # that in turn causes GHCi to abort
   stripDebugFlags = [ "-S" ] ++ stdenv.lib.optional (!targetPlatform.isDarwin) "--keep-file-symbols";
 
-  # zsh and other shells are smart about `{ghc}` but bash isn't, and doesn't
-  # treat that as a unary `{x,y,z,..}` repetition.
   postInstall = ''
-    paxmark m $out/lib/${name}/bin/${if targetPlatform != hostPlatform then "ghc" else "{ghc,haddock}"}
+    for bin in "$out"/lib/${name}/bin/*; do
+      isELF "$bin" || continue
```
Comment (Ericson2314) [inline pkgs/development/compilers/ghc/8.0.2.nix]: This should fix the build failure.

**Comment by peti** [comment]
I merged the code to `haskell-updates`. Hydra is building it starting with evaluation https://hydra.nixos.org/eval/1459088.

**Comment by Ericson2314** [comment]
@peti Thank you!

**Comment by Ericson2314** [comment]
Hydra looks pretty good. I'm merging!

**Pull request merged**